### PR TITLE
Move the nroff elves to GitHub actions

### PR DESCRIPTION
**Pull request by jsquyres**
This PR moves the Nroff Elves -- our lovable minions who faithfully render our Markdown-based documentation to nroff for us -- from a GitHub webhook executed on flatbed2.openfabrics.org to GitHub Actions.

## Summary

The Nroff Elves GitHub Action will fire for every PR.  It will effectively detect if any `man/*.md` files have changed, and if so, render new corresponding nroff man pages in `man/man?/*`, commit them, and then push the new commit to that PR.
    
This is a change compared to the old nroff-auto-generation scheme, which would see `man/*.md` changes *after* a PR was merged, and create a new PR with updated nroff-rendered man pages in `man/man?/*`, wait for CI to complete, and then merge that PR.
    
The rationale for this change is as follows:
    
1. This GitHub Action is *significantly* simpler than the old scheme.
2. GitHub Actions are both free and easy to maintain.
3. Moving to GitHub Actions for the Nroff Elves removes one more service from the flatbed2.openfabric.org server, which moves them one step towards eliminating that (expensive) server.
    
## Notes

The *first* time this GitHub Action runs (which will be *after* this PR merges to the main libfabric repo), it will cause a re-generation of *all* the existing man pages.  This is because the version of Pandoc used to generate the nroff is significantly newer on the GitHub Action Ubuntu container compared to the Pandoc version we're running on flatbed2.openfabrics.org.  Hence, the rendered nroff will change in all files in lots of little, effectively meaningless ways.  After that, we should see the usual/expected behavior of seeing OFIWG Bot commits from the Nroff Elves only when  `man/*.md` files change.

Additionally, as noted above, this new scheme will end up pushing a new commit to the same/existing PR.  This will end up re-firing all the other CI jobs on that PR.  It *should* be a pretty quick turnaround time (i.e., the Nroff Elves are fairly quick in rendering new nroff and pushing a new commit to the PR -- maybe within a minute or two? -- and then the CI jobs will effectively re-start).

## Example

If you want to see an example of this GitHub Action in action (see what I did there?), see https://github.com/jsquyres/libfabric/pull/52:

1. You'll see the same commits as are here on this PR.
1. You'll see the GitHub Action fire in CI on git commit f517193.
1. Additionally, because of some magic I did, you'll also see the "let's re-generate all the nroff with a newer Pandoc, resulting in lots of (meaningless) changes in the rendered nroff" commit on that PR.
1. There's also a commit which makes a small change to `man/fi_usnic.7.md`, just as an example.
1. You can then see the Nroff Elves dutifully added another commit to the PR with the re-rendered `man/man7/fi_usnic.7`.

### CODE REVIEW

File: config/md2nroff.pl [hunk 1]
```diff
@@ -132,8 +132,8 @@
     # compare and ignore if the date has changed.  Note that some
     # versions of pandoc render dates as xxxx\-xx\-xx, and others
     # render it as xxxx-xx-xx.  Handle both.
-    $target_nroff =~ s/\"\d\d\d\d\\\-\d\d\\\-\d\d\"/\"\\\@DATE\\\@\"/;
-    $target_nroff =~ s/\"\d\d\d\d\-\d\d\-\d\d\"/\"\\\@DATE\\\@\"/;
+    $target_nroff =~ s/\"\d\d\d\d\\\-\d\d\\\-\d\d\"/\"#DATE@\"/;
```
Comment (acgoldma) [inline config/md2nroff.pl:135]: Trailing "#"

Comment (jsquyres) [reply config/md2nroff.pl:135]: Good catch -- thank you!

**Comment by jsquyres** [comment]
NOTE: I think there will be some corner cases surrounding when the nroff generation is moved to a GitHub Action ("GHA").

I'm actually not sure what will happen to PRs that 1) are open when this PR is merged, and b) contain changes to `man/*.md` files.  Specifically, I _think_ the GH Action will only run on a PR that includes the GHA commit on the base branch (e.g., `master`).  Meaning:

* If you created a PR _before_ this GHA PR is merged, the GHA won't run on your PR.
    * Unless you rebase your PR to also include the GHA commit on the base (`master`) branch of your PR.
* This also affects PRs to release branches
    * Meaning: we might want to cherry pick these GHA commits to all active release branches

If your PR:

1. Updates `man/*.md`
2. Doesn't run the GHA to re-render the nroff (e.g., because your PR base branch does not include the GHA commit in its history on the base branch)
3. Merges (e.g., to `master`)

Then the *next* PR that includes your commits will trigger the Nroff Elves to re-render the nroff.  This may be unexpected for the author of that PR (because they didn't edit any `man/*.md` files).

These kinds of edge conditions will disappear over time, but there may/will be some awkwardness when we transition to the new Nroff Elves GHA.

**Comment by shefty** [comment]
I think we can live with that issue, but thanks for noting it.

**Comment by jsquyres** [comment]
> I think we can live with that issue, but thanks for noting it.

If you're cool with this PR, I'll merge.

You might want to rebase #6629 after this PR merges.

**Review by shefty — APPROVED** [review]

**Comment by shefty** [comment]
Yes, I'll rebase my PR.

**Pull request merged**